modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/plugin/ArtifactPluginDescriptorFactory.java

```diff
@@ -193,9 +195,16 @@ private ArtifactPluginDescriptor loadFromJsonDescriptor(File pluginFolder, File
       final ClassLoaderModel classLoaderModel =
           mavenClassLoaderModelLoader.loadClassLoaderModel(pluginFolder, classLoaderModelLoaderDescriptor.getAttributes());
       descriptor.setClassLoaderModel(classLoaderModel);
-      descriptor.setBundleDescriptor(mavenClassLoaderModelLoader.loadBundleDescriptor(pluginFolder));
     });
 
+    MuleArtifactLoaderDescriptor bundleDescriptorLoader = mulePluginModel.getBundleDescriptorLoader();
+    if (!bundleDescriptorLoader.getId().equals(MAVEN)) {
+      throw new ArtifactDescriptorCreateException(format("The identifier '%s' for a bundle descriptor loader is not supported (error found while reading plugin '%s')",
+                                                         bundleDescriptorLoader.getId(),
+                                                         pluginFolder.getAbsolutePath()));
+    }
+    descriptor.setBundleDescriptor(new MavenBundleDescriptorLoader().loadBundleDescriptor(pluginFolder));
```

Comment:
despite not being used, shouldn't we pass the `bundleDescriptorLoader.getAttributes()` to the `#loadBundleDescriptor(..)` method?

Also, as there is two implementations of the "bundle loader", maybe an interface where it gets the File+attributes is needed, and we would implement SPI for the `MAVEN` ID and the `PROPERTIES_BUNDLE_DESCRIPTOR_LOADER_ID` ID (as it happens in PropertiesBundleDescriptorLoader)